arangod/Agency/Agent.h

```diff
@@ -385,7 +386,7 @@ class Agent final : public arangodb::ServerThread<ArangodServer>,
   Constituent _constituent;
 
   /// @brief Cluster supervision module
-  Supervision _supervision;
+  std::unique_ptr<Supervision> _supervision;
```

Comment:
Out of interest: Why?

Reply:
This allows me to have this header included without making the implementation of Supervision visible. Not having it as a pointer would require every user of Agent to also include Supervision.h.